### PR TITLE
chore: Remove numpy pin, pin datashader, drop Python 3.7

### PR DESCRIPTION
**Pull request by jthetzel**
## Proposed Changes

  - Pin datashader version to 0.15.0 and remove numpy version pin.
  - Require Python >=3.8.

Xarray-spatial tests were failing due to datashader calling `numpy.warnings`, which is removed in numpy >=1.24.0 ( https://github.com/makepath/xarray-spatial/issues/779 ). Datashader 0.15.0 removes calls to `numpy.warnings` (pr: https://github.com/holoviz/datashader/pull/1176 , changelog: https://github.com/holoviz/datashader/releases/tag/v0.15.0). This pull requests removes the numpy <= 1.23.4 pin in favour of a datashader >= 1.15.0 pin.

Datashader requires Python >=3.8, so this also drops support for Python 3.7.

### CODE REVIEW

**Comment by jthetzel** [comment]
Datashader 0.15.0 only supports Python >= 3.8:
https://pypi.org/project/datashader/0.15.0/

**Comment by jthetzel** [comment]
https://github.com/makepath/xarray-spatial/issues/770 considers dropping Python 3.7 support upon addition of Python 3.11 support. I'll defer to the maintainers.

**Comment by jthetzel** [comment]
To accomodate Datashader 0.15.0, this pull request also drops Python 3.7 support.

**Comment by brendancol** [comment]
@jthetzel thanks so much for doing this. @thuydotm will check and merge

**Review by thuydotm — APPROVED** [review]

**Comment by thuydotm** [comment]
Thanks @jthetzel! The new changes look good to me, merging.

**Pull request merged**
